src/makerom/rom_header.h

```diff
@@ -0,0 +1,143 @@
+#ifndef ROM_HEADER_H
+#define ROM_HEADER_H
+
+/* Storage medium IDs, used internally in MEDIUM below */
+
+#define STORAGE_MEDIUM_CARTRIDGE            "N"
+#define STORAGE_MEDIUM_CARTRIDGE_EXPANDABLE "C"
+#define STORAGE_MEDIUM_DISK                 "D"
+#define STORAGE_MEDIUM_DISK_EXPANSION       "E"
+
+/* Region IDs, used internally in REGION below */
+
+#define REGION_CODE_ALL      "A"
+#define REGION_CODE_JP       "J"
+#define REGION_CODE_US       "E"
+#define REGION_CODE_PAL      "P"
+#define REGION_CODE_GATEWAY  "G"
+#define REGION_CODE_LODGENET "L"
+
+/**
+ * Magic value to determine if the ROM is byteswapped.
+ *
+ * This is not totally reliable since the PI Domain 1 Latency could also hold this value, however generally it can be
+ * assumed that this is not the case.
+ */
+#define ENDIAN_IDENTIFIER \
+    .byte 0x80
+
+/**
+ * Configures the timings for PI Domain 1. This determines how fast the PI hardware will read from the ROM. IPL2 reads
+ * this configuration using the slowest possible configuration before switching to the provided configuration.
+ */
+#define PI_DOMAIN_1_CFG(lat, pwd, pgs, rls)   \
+    .byte (((rls) & 3) << 4) | ((pgs) & 0xF); \
+    .byte (pwd) & 0xFF;                       \
+    .byte (lat) & 0xFF
+
+/**
+ * Some older libultra versions use this field to set osClockRate. It does not have any other meaningful effect, the
+ * clock rate of physical units does not actually change to match this value.
+ */
+#define SYSTEM_CLOCK_RATE_SETTING(num) \
+    .word (num)
+
+/**
+ * Indicates the entrypoint address of the program that IPL3 will load the boot segment to and execute.
+ * This should be >= 0x80000400.
+ */
+#define ENTRYPOINT(sym) \
+    .word (sym)
+
+/**
+ * Indicates the hardware revision the program is designed for (hw_major, hw_minor) and what libultra version (os_ver)
+ * it uses.
+ * The hardware revision for a retail N64 is (2,0).
+ * The libultra version may be a single letter, without quotes.
+ */
+#define LIBULTRA_VERSION(hw_major, hw_minor, os_ver)   \
+    .half 0;                                           \
+    .byte (hw_major) * 10 + (hw_minor);                \
+    _os_ver_start = .;                                 \
+    .ascii #os_ver ;                                   \
+    .if (. - _os_ver_start) != 1;                      \
+        .error "OS version should be just one letter"; \
+    .endif
+
+/**
+ * ROM Checksum value. IPL3 computes a checksum over ROM data in the range 0x1000 to 0x101000 and compares it to this
+ * value, if the results differ it will refuse to boot the program.
+ *
+ * This can generally be left as 0 in the ROM header declaration and filled in after the full ROM image is available to
+ * compute the checksum with.
+ */
+#define CHECKSUM(value)                 \
```

Comment:
it might not be obvious that this should be a 64-bit value (though I don't think anyone will put something other than 0 here anyway)

Reply:
Perhaps it would be better to just not have a value and just hardcode zeros, saying the checksum gets calculated at generation time?

Reply:
I don't think there's any especially valid case where it should be non-zero before proper calculation. There was one situation I was thinking of but I don't think it really needs to be supported out-of-the-box. I'll change it to take no args and just write 0 always.